$NetBSD: patch-sg,v 1.1 2007/11/29 02:49:12 rillig Exp $

For sunpro, "const char *" and "typedef'd const char *" are different
types that result in different mangled names. That's really hard to find
out:

Undefined symbol:
void UT_XML::processingInstruction(const char *, const char *);

Existing symbol:
void UT_XML::processingInstruction(const char *, const char *);

Spot the difference!

I couldn't find a tool that actually shows the differences between the
two mangled symbols. "dem --verbose __1c..." would be nice. No cookie
for Sun Microsystems.

--- src/af/util/xp/ut_xml.cpp.orig	2005-09-17 18:10:32.000000000 +0200
+++ src/af/util/xp/ut_xml.cpp	2007-11-28 17:19:04.574680000 +0100
@@ -174,9 +174,7 @@ void UT_XML::flush_all ()
     }
 }
 
-/* Declared in ut_xml.h as: void UT_XML::startElement (const XML_Char * name, const XML_Char ** atts);
- */
-void UT_XML::startElement (const char * name, const char ** atts)
+void UT_XML::startElement (const XML_Char * name, const XML_Char ** atts)
 {
   if (m_bStopped) return;
 
@@ -242,7 +240,7 @@ void UT_XML::charData (const char * buff
   m_chardata_buffer[m_chardata_length] = 0;
 }
 
-void UT_XML::processingInstruction (const char * target, const char * data)
+void UT_XML::processingInstruction (const XML_Char * target, const XML_Char * data)
 {
   if (m_bStopped) return;
   if (m_pExpertListener)
